lib/Cron/Workers/SimpleWorker.php

```diff
@@ -42,6 +42,11 @@ function init() {
 
   function process() {
     if (!$this->checkProcessingRequirements()) {
+      ScheduledTask::where('type', static::TASK_TYPE)
+        ->whereLte('scheduled_at', Carbon::createFromTimestamp($this->wp->currentTime('timestamp')))
+        ->whereNull('deleted_at')
+        ->whereNotEqual('status', ScheduledTask::STATUS_COMPLETED)
+        ->deleteMany();
```

Comment:
I just came across this and I have a question - could we reuse `$scheduled_tasks` and `$running_tasks` that are fetched just a few lines below and call `->delete()` on them? Because those two arrays are the exact tasks that cause the problems and because this is just a logic duplicity, moreover not a nice one since it's a SQL query, and not a nice one since the query is not targetting _exactly_ the same tasks.

We could fetch `$scheduled_tasks` and `$running_tasks` a bit earlier or save the `checkProcessingRequirements` to a variable and act later or both, to some extent.

Reply:
I thought about that. But Is it possible to call `delete()` on the result of `findMany()`? or do we have to loop over them and delete one by one?

Reply:
We can just loop over them, I think. In most cases there will probably be just one anyway.